SRC/PORT.CPP

```diff
@@ -196,9 +217,52 @@ static uint8_t tk_port_get_scancode_index( SDL_Keycode keycode, SDL_Scancode sca
     }
 }
 
+static void tk_port_synthetize_key_repeats()
+{
+    uint32_t now = SDL_GetTicks();
+
+    for ( auto &it : key_repeats )
+    {
+        SDL_Scancode scancode = it.first;
+        uint32_t last_repeat_ms = it.second.first;
+        int num_repeats = it.second.second;
+
+        uint32_t elapsed = now - last_repeat_ms;
+        if ( ( num_repeats == 0 && elapsed >= SYMBOL_KEY_REPEAT_START ) ||
+             ( num_repeats > 0 && elapsed >= SYMBOL_KEY_REPEAT_INTERVAL ) )
+        {
+            SDL_Event event;
+            event.type = SDL_KEYDOWN;
+            event.key.timestamp = now;
+            event.key.repeat = 1;
+            event.key.keysym.scancode = scancode;
+            SDL_PushEvent( &event );
+
+            it.second.first = now;
+            it.second.second = num_repeats + 1;
+        }
+    }
+}
+
+static void tk_port_init_key_repeat( SDL_Scancode scancode )
+{
+    uint32_t now = SDL_GetTicks();
+    if ( key_repeats.find( scancode ) == key_repeats.end() )
+    {
+        key_repeats.insert( std::pair< SDL_Scancode, std::pair< uint32_t, int > >( scancode, std::pair< uint32_t, int >( now, 0 ) ) );
```

Comment:
Simple `key_repeats.insert( { scancode, { now, 0 } } );` works in C++11. In case C++11 is not available, `std::make_pair` also allows implicit typing.